src/v/utils/process.cc

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "utils/process.h"
+
+#include "ssx/future-util.h"
+#include "vassert.h"
+#include "vlog.h"
+
+#include <seastar/core/loop.hh>
+#include <seastar/core/when_all.hh>
+#include <seastar/util/log.hh>
+
+#include <sstream>
+#include <string>
+
+static ss::logger proclog{"process"};
+
+namespace ssx {
+
+ss::future<>
+consume_input_stream(ss::input_stream<char> stream, bool is_stdout) {
+    ss::sstring stream_name{is_stdout ? "STDOUT" : "STDERR"};
+
+    while (!stream.eof()) {
+        auto buf = co_await stream.read();
+        if (buf.empty()) {
+            continue;
+        }
+
+        std::string str_buf{buf.begin(), buf.end()}, line;
+        std::stringstream ss(str_buf);
+        while (!ss.eof()) {
+            std::getline(ss, line);
+            vlog(proclog.trace, "{}: {}", stream_name, line);
+        }
+    }
+}
+
+process::~process() {
+    if (is_running()) {
+        vlog(
+          proclog.trace,
+          "Process still running, sending SIGKILL: cmd {}",
+          _cmd_str);
+        try {
+            _process->kill();
+        } catch (const std::system_error&) {
+            // If SIGKILL failed, then the process was already dead and it is
+            // safe to silently ignore.
+        }
+    }
+}
+
+ss::future<std::error_code> process::spawn(
+  const std::filesystem::path& cmd, ss::experimental::spawn_parameters params) {
+    gate_guard g{_gate};
```

Comment:
We should probably remove gate_guard from our codebase and just use `_gate.hold()`

Reply:
I'd be interested to see what others think

Reply:
I might tack it into a follow-up to my seastar PR